src/Microsoft.AspNetCore.JsonPatch/JsonPatchDocument.cs

```diff
@@ -213,6 +231,30 @@ public void ApplyTo(object objectToApplyTo, IObjectAdapter adapter)
             }
         }
 
+        /// <summary>
+        /// Apply this JsonPatchDocument
+        /// </summary>
+        /// <param name="objectToApplyTo">Object to apply the JsonPatchDocument to</param>
+        /// <param name="adapter">IObjectAdapterWithTest instance to use when applying</param>
+        public void ApplyTo(object objectToApplyTo, IObjectAdapterWithTest adapter)
```

Comment:
I think that the old `ApplyTo(object objectToApplyTo, IObjectAdapter adapter)` overload should test for the `adapter` param implementing `IObjectAdapterWithTest`, and if so, call the new overload so that legacy compiled code can benefit from the new Test operation without having to be recompiled.

@rynowak - any additional thoughts on this?

Reply:
That seems right to me

Reply:
I don't think we need an overload. The new interface implements the old one so it can use the older method. The two are handled differently in `Apply` method in `Operation` (Part of my update).